core/src/main/java/org/mskcc/cbio/portal/dao/DaoGenericAssay.java

```diff
@@ -143,4 +147,42 @@ public static boolean geneticEntitiesOnlyExistInSingleStudy(int cancerStudyId) t
         // do not update if there is an error
         return false;
     }
+
+    /**
+     * delete generic assay meta records if meta is not shared with other studies
+     * check and delete generic assay meta records profile by profile
+     * @throws DaoException
+     */
+    public static void checkAndDeleteGenericAssayMetaInStudy(int internalCancerStudyId) throws DaoException {
+        List<GeneticProfile> geneticProfiles = DaoGeneticProfile.getAllGeneticProfiles(internalCancerStudyId);
+        List<GeneticProfile> genericAssayProfiles = geneticProfiles.stream().filter(profile -> profile.getGeneticAlterationType() == GeneticAlterationType.GENERIC_ASSAY).collect(Collectors.toList());
+        List<Pair<Integer, String>> deleteGenericAssayStatements = new ArrayList<>();
+
+        for (GeneticProfile profile : genericAssayProfiles) {
+            if (DaoGenericAssay.geneticEntitiesOnlyExistInSingleStudy(profile.getGeneticProfileId())) {
```

Comment:
Profiles can have different generic entities so this has to be checked each iteration through the loop.